iocore/eventsystem/Thread.cc

```diff
@@ -72,42 +71,40 @@ init_thread_key()
 ///////////////////////////////////////////////
 
 struct thread_data_internal {
-  ThreadFunction f;
-  void *a;
-  Thread *me;
-  char name[MAX_THREAD_NAME_LENGTH];
+  ThreadFunction f;                  ///< Function to excecute in the thread.
+  Thread *me;                        ///< The class instance.
+  char name[MAX_THREAD_NAME_LENGTH]; ///< Name for the thread.
 };
 
 static void *
 spawn_thread_internal(void *a)
 {
-  thread_data_internal *p = (thread_data_internal *)a;
+  auto *p = static_cast<thread_data_internal *>(a);
 
   p->me->set_specific();
   ink_set_thread_name(p->name);
+
   if (p->f) {
```

Comment:
Now that the thread function is `ThreadFunction` value, can this condition ever be false?